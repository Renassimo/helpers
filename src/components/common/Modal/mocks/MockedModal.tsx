import { FormEvent, ReactNode } from 'react';
import { Breakpoint } from '@mui/material';

const MockedModal = jest.fn(
  ({
    open,
    children,
    onClose,
    onSubmit,
    title,
    loading,
    disabled,
    maxWidth = 'sm',
  }: {
    open: boolean;
    children: ReactNode;
    onClose: () => void;
    onSubmit: (event: FormEvent<HTMLFormElement>) => void;
    title: string;
    loading: boolean;
    disabled?: boolean;
    maxWidth?: Breakpoint;
  }) => (
    <div>
      {children}
      <div>
        title: {title}; maxWidth: {maxWidth};{open ? 'open' : 'closed'};
        {loading ? 'loading' : 'not loading'};
        {disabled ? 'disabled' : 'not disabled'}; onClose{' '}
        {!!onClose ? 'passed' : 'did not passed'}; onSubmit{' '}
        {!!onSubmit ? 'passed' : 'did not passed'};
      </div>
    </div>
  )
);

export default MockedModal;
