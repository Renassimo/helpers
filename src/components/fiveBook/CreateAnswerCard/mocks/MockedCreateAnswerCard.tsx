const MockedCreateAnswerCard = jest.fn(({ children }) => (
  <div>
    Mocked Create Answer Card:
    {children}
  </div>
));

export default MockedCreateAnswerCard;
