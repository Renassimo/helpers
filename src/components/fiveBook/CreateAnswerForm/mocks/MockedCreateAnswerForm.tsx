const MockedCreateAnswerForm = jest.fn(() => (
  <div>Mocked Create Answer Form</div>
));

export default MockedCreateAnswerForm;
