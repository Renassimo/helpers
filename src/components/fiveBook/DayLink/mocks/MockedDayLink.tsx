const MockedDayLink = jest.fn(({ prev, next }) => (
  <div>
    Mocked Day Link: Prev: {prev}
    Next: {next}
  </div>
));

export default MockedDayLink;
