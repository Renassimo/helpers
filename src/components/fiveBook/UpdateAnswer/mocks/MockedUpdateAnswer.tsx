const MockedUpdateAnswer = jest.fn(() => <div>Mocked Update Answer:</div>);

export default MockedUpdateAnswer;
