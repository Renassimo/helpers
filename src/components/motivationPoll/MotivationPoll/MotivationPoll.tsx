import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import PollForm from '@/components/motivationPoll/PollForm';
import Results from '@/components/motivationPoll/Results';

import Box from '@mui/material/Box';

const MotivationPoll = ({ onDownloadPdf }: { onDownloadPdf: () => void }) => {
  const { results } = useMotivationPoll();
  return (
    <Box>
      {results ? <Results onDownloadPdf={onDownloadPdf} /> : <PollForm />}
    </Box>
  );
};

export default MotivationPoll;
