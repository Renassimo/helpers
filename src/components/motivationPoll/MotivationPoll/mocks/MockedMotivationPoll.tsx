const MockedMotivationPoll = jest.fn(() => <div>MockedMotivationPoll</div>);

export default MockedMotivationPoll;
