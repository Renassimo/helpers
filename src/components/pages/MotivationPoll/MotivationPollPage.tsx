import MotivationPollProvider from '@/providers/motivationPoll/MotivationPollProvider';

import usePdfDownload from '@/hooks/common/usePdfDownload';

import { MotivationPollApiData } from '@/types/motivationPoll';

import PageTemplate from '@/components/templates/PageTemplate';
import MotivationPoll from '@/components/motivationPoll/MotivationPoll';
import LocaleLinks from '@/common/components/LocaleLinks';

const MotivationPollPage = ({
  data,
  locales,
  locale,
}: {
  data: MotivationPollApiData;
  locales: string[];
  locale: string;
}) => {
  const { pdfRef, onDownloadPdf } = usePdfDownload('motivation-poll-results');
  return (
    <MotivationPollProvider data={data}>
      <div ref={pdfRef}>
        <PageTemplate title="Motivation Test" pages={[]}>
          <MotivationPoll onDownloadPdf={onDownloadPdf} />
          <LocaleLinks
            locale={locale}
            locales={locales}
            href="/motivationTest"
          />
        </PageTemplate>
      </div>
    </MotivationPollProvider>
  );
};

export default MotivationPollPage;
