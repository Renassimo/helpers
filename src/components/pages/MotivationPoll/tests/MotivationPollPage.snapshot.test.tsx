import renderWithTheme from '@/tests/helpers';

import usePdfDownload from '@/hooks/common/usePdfDownload';
import MotivationPoll from '@/components/motivationPoll/MotivationPoll';
import MotivationPollPage from '../MotivationPollPage';
import { mockedApiData } from '@/types/motivationPoll/mocks';
import MockedMotivationPoll from '@/components/motivationPoll/MotivationPoll/mocks/MockedMotivationPoll';

jest.mock('@/hooks/common/usePdfDownload');
jest.mock('@/components/motivationPoll/MotivationPoll');

describe('MotivationPollPage snapshot', () => {
  const mockedOnDownloadPdf = jest.fn();
  const mockedPdfRef = null;
  const mockedUsePdfDownload = jest.fn(() => ({
    pdfRef: mockedPdfRef,
    onDownloadPdf: mockedOnDownloadPdf,
  }));

  beforeEach(() => {
    (usePdfDownload as jest.Mock).mockImplementation(mockedUsePdfDownload);
    (MotivationPoll as jest.Mock).mockImplementation(MockedMotivationPoll);
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { container } = renderWithTheme(
      <MotivationPollPage data={mockedApiData} />
    );
    // Assert
    expect(mockedUsePdfDownload).toHaveBeenCalledWith(
      'motivation-poll-results'
    );
    expect(container).toMatchSnapshot();
  });
});
