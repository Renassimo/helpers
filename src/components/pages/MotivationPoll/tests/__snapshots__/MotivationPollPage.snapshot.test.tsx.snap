// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MotivationPollPage snapshot renders successfully 1`] = `
<div>
  <div>
    <main
      class="css-3wo33r"
    >
      <div
        class="css-5aiikd"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-s31e50-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-9kk78p-MuiToolbar-root"
          >
            <div
              class="MuiBox-root css-0"
            />
            <div
              class="MuiBox-root css-0"
            />
          </div>
        </header>
      </div>
      <div
        class="css-16cpvj6"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
        >
          <div>
            MockedMotivationPoll
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
